assets/js/src/marketing_optin_block/block.tsx

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable react/react-in-jsx-scope */
+/* eslint-disable import/no-unresolved */
+
+/**
+ * External dependencies
+ */
+import { CheckboxControl } from '@woocommerce/blocks-checkout';
+import { useState, useEffect } from '@wordpress/element';
+import { getSetting } from '@woocommerce/settings';
+
+const { optinEnabled, defaultText, defaultStatus } = getSetting('mailpoet_data');
```

Comment:
I am not sure what would happen if this file gets loaded but the WooCommerce plugin is not active? Would this still work?

Reply:
The integration class is not initialized without WordPress and Blocks active, so this script would not be loaded.

Reply:
This code won't ever be reached if WooCommerce Blocks version 6.3.0 or up isn't enabled, so it's safe that way.
The block has a tight dependency on WooCommerce Checkout so it can't be used outside it if that's your question.